src/Stratis.Features.FederatedPeg/TargetChain/MaturedBlocksSyncManager.cs

```diff
@@ -112,6 +115,10 @@ protected async Task<bool> SyncBatchOfBlocksAsync(CancellationToken cancellation
                 // Log what we've received.
                 foreach (MaturedBlockDepositsModel maturedBlockDeposit in matureBlockDeposits)
                 {
+                    // Order transactions in block deterministically
+                    maturedBlockDeposit.Deposits = maturedBlockDeposit.Deposits.OrderBy(x => x.Id,
```

Comment:
In the specification we kept the original order of trx in a block is that not better?

Reply:
Unfortunately not - the opposing chain has no way to reason about the index of transactions in a block, but it's very easy for them to reason about canonical order.

Also if it was implemented like that, there are currently issues where deposits are arriving in different orders to different nodes, so it's not actually working as it should be.

Reply:
(And knowing the order of things is really important to pick transactions deterministically)

Reply:
>Also if it was implemented like that, there are currently issues where deposits are arriving in different orders to different nodes

This is odd once trx are in a block they maintain the order (that's crucial to calculate the merkel root)

Reply:
Hmm, there could have been a reorg on the counter chain. I'll try with and without this.

Reply:
There can not be reorgs because we wait for max reorg before we act on a transfer

Reply:
(not on the testnet tool)

Reply:
OK I think this is actually correct as the spec says
>If a block has more than one UTXO, then they are selected in order of transaction id.

This code only orders the trx in a block